### ISSUE TITLE
[BUG] npm update changing package.json

### ISSUE THREAD

**Issue by mattpen** [npm/cli#3044]
### Current Behavior:

`npm update` will make changes to package.json, even if `--save=false` or `--no-save` is present. This includes adding a newline to the end of package.json even if no other changes were present.

### Expected Behavior:

If save is set to false, `npm update` should not modify package.json

### Steps To Reproduce:

I see this same behavior with `npm config set save false`, `npm update --no-save`, and `npm update --save=false`.

```
$ echo -n {} | cat > package.json
$ cat package.json               
{}%                                                                                                  
$ npm config get save            
false
$ npm update                     

up to date, audited 1 package in 151ms

found 0 vulnerabilities
$ cat package.json               
{}
$
```

### Environment:

- OS: MacOS 11.2.3 (M1)
- Node: 15.9.0
- npm: 7.5.3

**Comment by ZwapKillrath** [comment]
Related to #708?

**Comment by darcyclarke** [comment]
@mattpen can't seem to reproduce; Can you try updating to latest (`npm i -g npm` should be `v7.9.0`)

**Comment by mattpen** [comment]
@darcyclarke - I just upgraded and I still get the same results (npm update inserts trailing newline into package.json) with v7.9.0.

```
$ echo -n {} | cat > package.json
$ cat package.json
{}%
$ npm config get save
false
$ npm update

up to date, audited 1 package in 158ms

found 0 vulnerabilities
$ cat package.json 
{}
$ npm -v
7.9.0
$ 
```

**Comment by mattpen** [comment]
Using node 14 or 16 this problem does not occur.

**Comment by ruyadorno** [comment]
https://github.com/npm/cli/pull/4223 cleans up the usage of `save` config in `npm update`

Closing this issue since the original problem should be fixed in `npm@8.3.2`
